src/Region.cc

```diff
@@ -73,15 +90,63 @@ std::unique_ptr<Region> Region::decode(std::uint8_t const * buffer, size_t n) {
     throw std::runtime_error("Byte-string is not an encoded Region");
 }
 
+std::unique_ptr<Region> Region::decodeBase64(std::string_view const & s) {
+    auto region_begin = s.begin();
+    auto region_end = std::find(s.begin(), s.end(), ':');
+    if (region_end != s.end()) {
+        std::vector<std::unique_ptr<Region>> union_args;
+        while (region_end != s.end()) {
+            auto bytes = base64::decode_into<std::vector<std::uint8_t>>(region_begin, region_end);
+            union_args.push_back(decode(bytes));
+            region_begin = region_end;
+            ++region_begin;
+            region_end = std::find(region_begin, s.end(), ':');
+        }
+        auto bytes = base64::decode_into<std::vector<std::uint8_t>>(region_begin, region_end);
+        union_args.push_back(decode(bytes));
+        return std::unique_ptr<UnionRegion>(new UnionRegion(std::move(union_args)));
+    } else {
+        auto bytes = base64::decode_into<std::vector<std::uint8_t>>(region_begin, region_end);
+        return decode(bytes);
```

Comment:
My guess is that in this scheme an empty UnionRegion is represented by an empty string? I'm not sure that it could happen in our daf_butler queries, but it may be worth to support it in general, and this code will probably throw an exception for an empty string?